internal/services/migrate_rewards.go

```diff
@@ -0,0 +1,107 @@
+package services
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	pb_defs "github.com/DIMO-Network/device-definitions-api/pkg/grpc"
+	"github.com/DIMO-Network/rewards-api/models"
+	"github.com/DIMO-Network/shared/db"
+	"github.com/ericlagergren/decimal"
+	"github.com/rs/zerolog"
+	"github.com/volatiletech/sqlboiler/v4/types"
+)
+
+func MigrateRewardsController(ctx context.Context, logger *zerolog.Logger, pdb db.Store, allIntegrations *pb_defs.GetIntegrationResponse, week int) error {
+
+	integrsByID := make(map[string]*pb_defs.Integration)
+	for _, integr := range allIntegrations.Integrations {
+		integrsByID[integr.Id] = integr
+	}
+
+	rewards, err := models.Rewards(
+		models.RewardWhere.IssuanceWeekID.LT(week),
```

Comment:
I think probably `.LTE` would be better? "I want to migrate this week and all weeks before it".